frontend/packages/dev-console/src/components/edit-application/edit-application-types.ts

```diff
@@ -0,0 +1,24 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResult } from '@console/internal/components/utils';
+import { NormalizedBuilderImages } from '../../utils/imagestream-utils';
+
+export interface EditApplicationFormProps {
+  builderImages?: NormalizedBuilderImages;
+}
+
+export interface AppResources {
+  service?: FirehoseResult;
+  route?: FirehoseResult;
+  buildConfig?: FirehoseResult;
+  imageStreams?: FirehoseResult;
+}
+
+export interface EditApplicationProps {
+  namespace: string;
+  appName: string;
+  resource: K8sResourceKind;
+  resources?: AppResources;
+  loaded?: boolean;
+  cancel?: () => void;
+  close?: () => void;
```

Comment:
I much prefer the convention of using prefix `on`
`onSubmit` and `onCancel`

Rename `close` because you aren't closing the form. You are notifying the parent the form is submitted. If it happens to be in a modal and the parent wants to close the modal, then it's up the parent. But the form itself isn't closing.